lib/dartino/launch_dartino.dart

```diff
@@ -113,4 +115,56 @@ class DartinoLaunch extends Launch {
     if (result != 0 || isLast) launchTerminated(result, quiet: true);
     return result;
   }
+
+  /// Start the debugging session and return `true` if successful.
+  Future<bool> debug(DartinoSdk sdk, [String ttyPath]) async {
+    String command = sdk.dartinoBinary;
+    List<String> args = ['debug', 'serve', primaryResource];
+    if (ttyPath != null) args.addAll(['on', 'tty', ttyPath]);
+
+    pipeStdio('Starting debug session...\n');
+    pipeStdio('\$ $command ${args.join(' ')}\n', highlight: true);
+    runner = new ProcessRunner(command,
+        args: args, cwd: launchConfiguration.projectPath);
+
+    // Wait for the observatory port
+    Completer<int> portCompleter = new Completer<int>();
+    runner.onStdout.listen((str) {
+      if (str.startsWith('localhost:')) {
+        try {
+          portCompleter.complete(int.parse(str.substring(10)));
+        } catch (e) {
+          pipeStdio('Failed to parse observatory port from "$str"',
+              error: true);
+        }
+      }
+      pipeStdio(str, subtle: true);
+    });
+    runner.onStderr.listen((str) => pipeStdio('\n$str\n', error: true));
+    runner.execStreaming().then((int exitCode) {
+      pipeStdio('debug session exit code is $exitCode', highlight: true);
+      launchTerminated(exitCode, quiet: true);
+    });
+    int observatoryPort = await portCompleter.future
+        .timeout(new Duration(seconds: 5), onTimeout: () => null);
```

Comment:
Don't know yet. I picked 5 seconds as a way to exit out of the process if the port is not found in a reasonable amount of time, but perhaps I could ask the user if they want to continue waiting or ? I'll add a note here to ponder and follow up with a better solution.